test/cppsuite/tests/background_compact.cpp

```diff
@@ -0,0 +1,362 @@
+                    added_count = 0;
+                    tw->txn.rollback();
+                } else {
+                    added_count++;
+                    if (tw->txn.can_commit()) {
+                        if (tw->txn.commit()) {
+                            /*
+                             * We need to inform the database model that we've added these keys as
+                             * some other thread may rely on the key_count data. Only do so if we
+                             * successfully committed.
+                             */
+                            cc.coll.increase_key_count(added_count);
+                        } else {
+                            added_count = 0;
+                        }
+                    }
+                }
+
+                /* Sleep the duration defined by the op_rate. */
+                tw->sleep();
+            }
+            /* Reset our cursor to avoid pinning content. */
+            testutil_check(cc.cursor->reset(cc.cursor.get()));
+            counter++;
+            if (counter == collections_per_thread)
+                counter = 0;
+            testutil_assert(counter < collections_per_thread);
+        }
+        /* Make sure the last transaction is rolled back now the work is finished. */
+        tw->txn.try_rollback();
+    }
+
+    void
+    background_compact_operation(thread_worker *tw) override final
+    {
+        logger::log_msg(
+          LOG_INFO, type_string(tw->type) + " thread {" + std::to_string(tw->id) + "} commencing.");
+
+        bool enabled = false;
+
+        while (tw->running()) {
+            enabled = !enabled;
+
+            std::string compact_cfg = enabled ?
+              "background=true,free_space_target=" + std::to_string(tw->free_space_target_mb) +
+                "MB" :
+              "background=false";
+
+            auto ret = tw->session->compact(tw->session.get(), nullptr, compact_cfg.c_str());
+            if (ret == EBUSY) {
+                logger::log_msg(LOG_WARN,
+                  type_string(tw->type) + " thread {" + std::to_string(tw->id) +
+                    "}: background compact returned EBUSY.");
+                /* Toggle back to the original state if we failed to call background compact. */
+                enabled = !enabled;
+            } else
+                testutil_check(ret);
+
+            tw->sleep();
+        }
+    }
+
+    void
+    validate(bool, const std::string &, const std::string &, database &) override final
+    {
+        logger::log_msg(LOG_INFO, "Starting validation.");
+        scoped_session session = connection_manager::instance().create_session();
+
+        /* Check the background compact statistics. */
+        int64_t bytes_recovered, bytes_rewritten_ema, bytes_written, files_tracked, skipped,
+          success;
+        scoped_cursor conn_stat_cursor = session.open_scoped_cursor(STATISTICS_URI);
+
+        metrics_monitor::get_stat(
```

Comment:
I think this is a relatively clean way of doing this.

Reply:
Great! And is this in context to the discussion about validating stats in the cppsuite in general?

Reply:
Yeah it is though it does bring into question the purpose of the metrics monitor.

Reply:
I think that can be a discussion for [WT-11742](https://jira.mongodb.org/browse/WT-11742)